src/lib/github/proxy.ts

```diff
@@ -0,0 +1,100 @@
+import { NextRequest, NextResponse } from 'next/server';
+import { githubApiPath, githubApiUrl } from '@/lib/github/constants';
+import { requireAuth } from '@/lib/auth/requireAuth';
+import { getBearerAccessToken } from '@/lib/auth/getBearerAccessToken';
+import * as Response from '@/lib/response';
+import { fetchRequest } from '@/lib/request';
+
+export async function proxy<T>(
+  request: NextRequest
+): Promise<NextResponse<Result<T>>> {
+  try {
+    const isAuthenticated = await requireAuth();
+    if (!isAuthenticated) return Response.Unauthorized;
+
+    const bearerToken = await getBearerAccessToken();
+    if (!bearerToken) return Response.Unauthorized;
+
+    // Security check: Only allow POST for GraphQL requests
+    const method = request.method;
+    if (method === 'POST') {
+      const requestUrl = new URL(request.url);
+      const pathname = requestUrl.pathname;
```

Comment:
Not a priority but you could destructure the object?
```suggestion
      const { pathname }= requestUrl;
```